lib/util/package-rules/repository-patterns.ts

```diff
@@ -0,0 +1,44 @@
+import is from '@sindresorhus/is';
+import type { PackageRule, PackageRuleInputConfig } from '../../config/types';
+import { logger } from '../../logger';
+import { regEx } from '../regex';
+import { Matcher } from './base';
+import { massagePattern } from './utils';
+
+export class RepositoryPatternsMatcher extends Matcher {
+  override matches(
+    { repository }: PackageRuleInputConfig,
+    { matchRepositoryPatterns }: PackageRule
+  ): boolean | null {
+    if (is.undefined(matchRepositoryPatterns)) {
+      return null;
+    }
+
+    if (is.undefined(repository)) {
+      return false;
+    }
+
+    const reposToMatchAgainst = [repository];
```

Comment:
this array makes no sense

Reply:
I agree, it's also in the other pattern matchers. 🙃🙃

Reply:
they don't need to be same. code should be as easy as possible.

Reply:
I agree, wasn't making that argument 🙃

Reply:
Will correct shortly